src/cmd/src/cli/export.rs

```diff
@@ -120,18 +120,34 @@ pub struct Export {
 }
 
 impl Export {
+    fn catalog_path(&self) -> PathBuf {
+        PathBuf::from(&self.output_dir).join(&self.catalog)
+    }
+
     async fn get_db_names(&self) -> Result<Vec<String>> {
-        if let Some(schema) = &self.schema {
-            Ok(vec![schema.clone()])
-        } else {
-            self.all_db_names().await
-        }
+        let db_names = self.all_db_names().await?;
+        let Some(schema) = &self.schema else {
+            return Ok(db_names);
+        };
+
+        // Check if the schema exists
+        db_names
+            .into_iter()
+            .find(|db_name| db_name.to_lowercase() == schema.to_lowercase())
+            .map(|name| vec![name])
+            .context(SchemaNotFoundSnafu {
+                catalog: &self.catalog,
+                schema,
+            })
```

Comment:
Nit:
try this
```rust
use itertools::Itertools;

let db_names = self.all_db_names().await?;
let db_names = db_names
    .cartesian_product(self.schema.iter())
    .find_map(|(x, y)| if x.to_lowercase() == y.to_lowercase() {
        Some(x)
    } else {
        None
    })
    .collect();
ensure!(!db_names.is_empty(), SchemaNotFoundSnafu {
    catalog: &self.catalog,
    schema,
};

Reply:
Thank you for the suggestion, but my idea is to keep it simple since performance is not a major concern here.